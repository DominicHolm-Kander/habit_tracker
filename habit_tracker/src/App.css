.app {
  text-align: center;
  background-size: cover;
  height: 100%;
  width: 100%;
  color:black;
}
h1{
  font-family: 'Bangers', cursive;
}