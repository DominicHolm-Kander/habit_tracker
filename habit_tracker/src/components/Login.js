import React, { Component } from 'react'

export class Login extends Component {

    handleChangeOfPassword = (event) => {
        this.setState({currenInputPassword: event.target.value})
    }
    handleChangeOfUsername = (event) => {
        this.setState({currenInputUsername: event.target.value})
    }


    render() {


        return (
            <div>
                <h3>Put your login info here</h3>
                <label for="userName">Username
                <input type="text" className = "userName" name="userName" onChange={(event) => this.handleChangeOfUsername(event)}></input>
                </label>
                <label for="password">Password
                <input type="text" className = "password" name="password" onChange={(event) => this.handleChangeOfPassword(event)}></input>
                </label>
                <button type="button" id="loginButton" onClick={() => this.props.checkCredentials(this.state.currenInputUsername, this.state.currenInputPassword)}>Log in</button>
            </div
        )
    }
}

export default Login
