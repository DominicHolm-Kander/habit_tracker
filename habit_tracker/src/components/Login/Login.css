.login {
    display: flex;
    flex-direction: row;
}

.loginButton {
    outline: none;
    height: 50px;
    width: 100px;
    box-sizing: border-box;
    appearance: none;
    background-color: transparent;
    border-radius: 0.6em;
    cursor: pointer;
    display: flex;
    align-self: center;
    font-size: 1rem;
    font-weight: 400;
    line-height: 1;
    margin: 5px;
    padding: 1.2em 1.2em;
    text-decoration: none;
    text-align: center;
    text-transform: uppercase;
    font-family: 'Montserrat', sans-serif;
    font-weight: 700;
    border-color: rgb(0, 255, 64);
    color: #fff;
    box-shadow: 0 0 40px 40px rgb(39, 182, 26) inset, 0 0 0 0 rgb(20, 214, 69);
    transition: all 150ms ease-in-out;
}
  .loginButton:hover {
    box-shadow: 0 0 10px 0 rgb(47, 201, 33) inset, 0 0 10px 4px rgb(30, 189, 91);
    color: green;
}

