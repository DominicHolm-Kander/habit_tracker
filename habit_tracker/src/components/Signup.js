import React, { Component } from 'react'

export class Signup extends Component {

    handleChangeOfUsername = (event) => {
        this.setState({username: event.target.value})
    }
    handleChangeOfPassword = (event) => {
        this.setState({password: event.target.value})
    }
    handleChangeOfFn = (event) => {
        this.setState({fn: event.target.value})
    }
    handleChangeOfLn = (event) => {
        this.setState({ln: event.target.value})
    }
    

    render() {
        return (
            <div>
                <h3>Or if you're a first-time visitor, sign up below!</h3>
                <form>
                    <label for="firstName">First Name
                    <input type="text" className = "firstName" name="fn"onChange={(event) => this.handleChangeOfFn(event)}></input>
                    </label>
                    <label for="lastName">Last Name
                    <input type="text" className = "lastName" name="ln"onChange={(event) => this.handleChangeOfLn(event)}></input>
                    </label>
                    <label for="userName">Username
                    <input type="text" className = "userName" name="userName" onChange={(event) => this.handleChangeOfUsername(event)}></input>
                    </label>
                    <label for="password">Password
                    <input type="text" className = "password" name="password"onChange={(event) => this.handleChangeOfPassword(event)}></input>
                    </label>
                    <button type="button" id="signUpButton" onClick={() => this.props.signup(this.state)}>Sign up</button>
                </form>
            </div>
        )
    }
}

export default Signup
