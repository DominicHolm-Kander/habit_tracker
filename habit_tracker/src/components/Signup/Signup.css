.btn{
    margin-top:40px;
    font-family: 'Bowlby One SC', cursive;
}
.col-md-4, .mb-3{
    margin-top:500px;
}
h4{
    margin-left:20px;
    margin-top:40px;
    font-family:'Bangers', cursive;
}
body{
    background-color: greenyellow;
}
/* .signup{
    margin-top:80px;
} */