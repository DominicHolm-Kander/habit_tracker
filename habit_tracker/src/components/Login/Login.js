import React, { Component } from 'react'
import './Login.css'
// import { FormControl } from '@material-ui/core';

export class Login extends Component {

    handleChangeOfPassword = (event) => {
        this.setState({currenInputPassword: event.target.value})
    }
    handleChangeOfUsername = (event) => {
        this.setState({currenInputUsername: event.target.value})
    }


    render() {


        return (
            <div className="login">
            <h3>Sign in</h3>
            <div className="container">
        <div className="form-group">
            <label for="exampleInputEmail1">Username</label>
            <input type="text" className="form-control" id="exampleInputEmail1" aria-describedby="emailHelp" placeholder="Username" onChange={(event) => this.handleChangeOfUsername(event)}></input>
        </div>
        <div className="form-group">
            <label for="exampleInputPassword1">Password</label>
            <input type="password" className="form-control" id="exampleInputPassword1" placeholder="Password" onChange={(event) => this.handleChangeOfPassword(event)}></input>
            <small id="emailHelp" className="form-text text-muted">We'll never share your password with anyone else.</small>
        </div>
        </div>
        <button type="submit" className="btn btn-primary" onClick={() => this.props.checkCredentials(this.state.currenInputUsername, this.state.currenInputPassword)}>Submit</button>
       
            </div>
        )
            
        }    
    
    }

export default Login
