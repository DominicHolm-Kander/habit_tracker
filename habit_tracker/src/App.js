import React, { Component } from 'react';
import { BrowserRouter as Router , Switch, Route } from 'react-router-dom';
import Login from './components/Login/Login';
import Signup from './components/Signup/Signup';
import axios from 'axios';
import AccountOverview from './containers/AccountOverview';
import './App.css';


export class App extends Component {

  constructor(props){
    super(props);
    this.state = {users: []}
  }

  getAllUsers(){
    axios.get('http://localhost:4000/users')
    .then((res) => {
      // const usersInfo = res.data;
      this.setState({
        users: res.data
      })
    })
    .catch((err) => {
      console.log('this is the error: ' + err)
    })
  }

  

  componentDidMount(){
    this.getAllUsers();
  }
  
  checkCredentials = (inputUsername, inputPassword) => {
    axios.get('http://localhost:4000/users/' + inputUsername)
    .then((result) => {
      this.setState({enteredDetails: result.data})
      if (result.data[0].password === inputPassword) {
        this.setState({currentUser: result.data[0]})
      }
    })
  }

  signupNewUser = (submittedDetails) => {
    console.log(submittedDetails)
    axios.post('http://localhost:4000/users/add', {
      fn: submittedDetails.fn,
      ln: submittedDetails.ln,
      username: submittedDetails.username,
      password: submittedDetails.password
    })
    .then(this.setState({currentUser: submittedDetails}))
  }

  render() {
    return (
      <Router>
        <div className='app'>
          <h1>Welcome to your favorite habit tracker!</h1>
          {!this.state.currentUser ? <Login checkCredentials={this.checkCredentials}/> : null}
          {!this.state.currentUser ? <Signup signup={this.signupNewUser}/>: null}
          {this.state.currentUser ? <AccountOverview  userInfo = {this.state.currentUser}/> : 'not currently logged in' }
        </div>
      </Router>
    )
  }
}

export default App
