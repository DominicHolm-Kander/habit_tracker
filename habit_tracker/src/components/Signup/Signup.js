import React, { Component } from 'react'
import './Signup.css'
export class Signup extends Component {

    handleChangeOfUsername = (event) => {
        this.setState({username: event.target.value})
    }
    handleChangeOfPassword = (event) => {
        this.setState({password: event.target.value})
    }
    handleChangeOfFn = (event) => {
        this.setState({fn: event.target.value})
    }
    handleChangeOfLn = (event) => {
        this.setState({ln: event.target.value})
    }
    

    render() {
        return (
          <div>
            <h4>If you're a first-time visitor, sign up below!</h4>
    <div className="signup">
            <form clasName="needs-validation" novalidate>
              <div clasName="form-row">
                <div clasName="col-md-4 mb-3">
                  {/* <label for="validationCustom01">First name</label> */}
                  <input type="text" clasName="form-control" id="validationCustom01" placeholder="First name" name="fn" required onChange={(event) => this.handleChangeOfFn(event)}></input>
                  
                </div>
              </div>
              <div clasName="col-md-4 mb-3">
                {/* <label for="validationCustom02">Last name</label> */}
                <input type="text" clasName="form-control" id="validationCustom02" placeholder="Last name" name="ln" onChange={(event) => this.handleChangeOfLn(event)} required></input>
                </div>
              
              <div clasName="col-md-4 mb-3">
                {/* <label for="validationCustom02">Username</label> */}
                <input type="text" clasName="form-control" id="validationCustom02" placeholder="Username" name="userName" onChange={(event) => this.handleChangeOfUsername(event)} required></input>
                </div>
              <div clasName="col-md-4 mb-3">
                {/* <label for="validationCustom02">Password</label> */}
                <input type="password" clasName="form-control" id="validationCustom03" placeholder="Password" name="password" onChange={(event) => this.handleChangeOfPassword(event)} required></input>
                </div>
              
             
              <div clasName="form-row">
                <div clasName="col-md-6 mb-3">
                  <button className="btn btn-primary" type="submit" onClick={() => this.props.signup(this.state)}>Sign up</button>
                </div>
              </div>
            </form>
            </div>
          </div>
        )
    }
}

export default Signup
