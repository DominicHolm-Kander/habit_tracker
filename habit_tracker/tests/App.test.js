import React from 'react'
import { shallow } from 'enzyme'
import App from '../src/app'
import Signup from '../src/components/Signup/Signup'
import Login from '../src/components/Login/Login'
import AccountOverview from '../src/containers/AccountOverview'
import waitUntil from "async-wait-until"; // For async testing of setState functions.
import regeneratorRuntime from 'regenerator-runtime';

describe("shallow app", () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<App />)
    })
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
    it("Should render signup and login components if no user currently logged in", () => {
        expect(wrapper.containsMatchingElement(<Signup />)).toEqual(true);
        expect(wrapper.containsMatchingElement(<Login />)).toEqual(true);
    })
    it("should not render signup and login component if a current user is logged in", 
    async () => {
        wrapper.setState({currentUser: 'text'})
        await waitUntil(() => wrapper.instance().state.currentUser)
        expect(wrapper.containsMatchingElement(<Signup />)).toEqual(false);
        expect(wrapper.containsMatchingElement(<Login />)).toEqual(false);
        expect(wrapper.containsMatchingElement(<AccountOverview />)).toEqual(true);
    })

});


