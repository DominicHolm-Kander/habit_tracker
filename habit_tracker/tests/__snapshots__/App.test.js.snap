// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow app should render correctly 1`] = `
<Component>
  <div
    className="app"
  >
    <h1>
      Welcome to your favorite habit tracker!
    </h1>
    <Login
      checkCredentials={[Function]}
    />
    <Signup
      signup={[Function]}
    />
  </div>
</Component>
`;
